src/stubs.ts

```diff
@@ -83,13 +83,16 @@ export const createStub = ({
   const anonName = 'anonymous-stub'
   const tag = name ? `${hyphenate(name)}-stub` : anonName
 
-  const propsDeclaration = type
-    ? unwrapLegacyVueExtendComponent(type).props || {}
+  const componentOptions = type
+    ? unwrapLegacyVueExtendComponent(type) || {}
     : {}
 
   return defineComponent({
     name: name || anonName,
-    props: propsDeclaration,
+    props: componentOptions.props || {},
+    // fix #1550 - respect old-style v-model for shallow mounted components with @vue/compat
+    // @ts-ignore
+    model: componentOptions.model,
```

Comment:
Did not know `model` was a thing - what is the TS error you get without `ts-ignore`? Ideally we don't ignore things, although maybe it's unavoidable. You could use `// @ts-expect-error` perhaps, since this is something we expect, not necessary an incorrect type, as such.

Reply:
`componentOptions` no longer includes the type for the `model` property, but it is there if you're using a Vue 2 component in Vue 2 mode. I have a patch locally using patch-package which basically does the same thing:

![image](https://user-images.githubusercontent.com/5778364/171939455-b3b7cece-3551-479e-87c4-8a475ad61449.png)

I can change it to expect-error if you would prefer

Reply:
I think that might be good - just to communicate why we have it a little better. Then we can merge!